src/rewards/rewards.service.ts

```diff
@@ -47,6 +51,38 @@ export class RewardsService {
     return claim;
   }
 
+  async list({
+    chain,
+    pageNum = 0,
+    pageCount = DEFAULT_PAGE_SIZE,
+  }: {
+    chain: Chain;
+    pageNum?: number;
+    pageCount?: number;
+  }): Promise<{
+    count: number;
+    records: UserClaimSnapshot[];
+  }> {
+    const { chainStartBlock, count } = await getLatestMetadata(chain);
+    const mapper = getDataMapper();
+    const records = [];
+    const startingPageId = pageNum * pageCount;
+    const endingPageId = startingPageId + pageCount - 1;
+    const expression: ConditionExpression = {
```

Comment:
nit: you could use `between(startingPageId, endingPageId)` instead using the annotations library